### PR TITLE
First draft of running the I18n Sync Down & Out periodically throughout the week.

### PR DESCRIPTION
**Pull request by Hamms**
This ended up being a lot simpler than I expected!

Specifically, we ended up just needing to make a couple tweaks to how the server manages its branches. Previously, the server would create the i18n sync branches during the i18n sync and then switch back to staging so it could continue to build. This means we are effectively "rewinding" the state of the server after the i18n sync, which would cause problems if we attempt to do more syncs before merging the sync PRs.

What we do now is simply stay on the i18n sync branch until we've detected that the branch has been merged into staging and _only then_ do we switch back to staging. If all goes well, this should mean that the server is only ever moving "forwards" in git history. This is somewhat more fragile than the setup we had before, so it could very well break if things don't go well, but because the i18n server is fairly isolated I'd suggest we deal with that potential chaos as it arises, rather than trying to anticipate.

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-1193)

## Testing story

I tested this out manually and added a unit test for the new github functionality, but this will still need some close monitoring once it gets merged.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: experimental/i18n-dev.crontab [hunk 1]
```diff
@@ -19,9 +19,25 @@
 PATH=/home/ubuntu/.rbenv/shims:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
 HOME=/home/ubuntu
 
+# Run a CI Build to update the server from staging. We do this primarily to
+# make sure the server gets seeded with latest level content for the sync in.
+# Note that the CI_ONLY_BUILD environment variable is necessary; the server is
+# not set up to actually serve a version of the website, and it will fail to
+# build correctly if not thus restricted.
 0 */4 * * * cd /home/ubuntu/code-dot-org && CI_ONLY_BUILD=true bundle exec ./bin/cronjob ./aws/ci_build >>/tmp/cronlog 2>&1
```
Comment (bethanyaconnor) [inline experimental/i18n-dev.crontab:27]: Will this run if we're not on the staging branch? Notably, I'm worried about a situation where we run the sync on Tuesday, never switch back to the staging branch, then the Monday sync is several days out of date.

Comment (Hamms) [reply experimental/i18n-dev.crontab:27]: oh, whoops I forgot to commit that change! One moment

Comment (bethanyaconnor) [reply experimental/i18n-dev.crontab:27]: Ahh the continuous sync doesn't switch branches. That still means it won't be able to pull from staging if there are conflicting files but I guess that's just something to look out for.

Comment (Hamms) [reply experimental/i18n-dev.crontab:27]: That's correct! The idea is that we progress through the branches like this:

1. Start on staging branch. From here, we will run the CI build to update our sources files from staging latest, and also run the continuous sync to update our translation files from Crowdin.
2. Sunday evening, we run the I18n sync, which switches us to the i18n sync branch. From here we can continue to run the continuous sync, but won't run the CI build.
3. Once the sync PRs have been merged, we switch back to the staging branch, which now includes the latest sync changes.

Comment (bethanyaconnor) [reply experimental/i18n-dev.crontab:27]: Thanks for the explanation! And of course the i18n down/out steps don't change any English files so the chance of conflict is low.

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**